qemu/tests/microcode_test.py

```diff
@@ -0,0 +1,46 @@
+import re
+import logging
+
+from avocado.utils import process
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Microcode test:
+    1) Get microcode version on host
+    2) Boot guest with '-cpu host'
+    3) Check if microcode version inside guest match host
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    def get_microcode_ver(cmd, session=None):
+        """
+        Get microcde version in guest or host
+        """
+        if session:
+            output = session.cmd_output(cmd)
+        else:
+            output = process.getoutput(cmd, shell=True)
+        ver = re.findall(r":\s?(0x[0-9A-Fa-f]+)", output)[0]
```

Comment:
How about `r":\s*(0x[0-9A-Fa-f]+` ? replace `?` with `*`, to cover 0, 1 and more space.

Reply:
It seems not right with a single '0x' but no any hexadecimal digits.

Reply:
Oh, I got you. Updated.